src/angular-app/bellows/shared/sound-player.component.ts

```diff
@@ -60,38 +67,67 @@ export class SoundController implements angular.IController {
         this.togglePlayback();
       }
 
+      this.audioElement.load();
       this.audioElement.src = urlChange.currentValue;
     }
   }
 
   $onDestroy(): void {
-    this.audioElement.pause();
+    if (!this.audioElement.paused){
+      this.audioElement.pause();
+    }
   }
 
   iconClass(): string {
     return this.playing ? 'fa-pause' : 'fa-play';
   }
 
+
+  async playAudio() {
+    try{
+      let loadedAudioPlayer = await this.audioElement.play();
+      return loadedAudioPlayer;
+    } catch (e) {
+
+    }
+  }
+
   togglePlayback(): void {
     this.playing = !this.playing;
 
     if (this.playing) {
-      this.audioElement.play();
+      this.audioElement.currentTime = 0;
+      this.playAudio();
     } else {
-      this.audioElement.pause();
+      if(!this.audioElement.paused){
+        this.audioElement.pause();
+      }
     }
+
   }
 
   currentTimeInSeconds(): number {
     return this.audioElement.currentTime;
   }
 
   durationInSeconds(): number {
-    return this.audioElement.duration;
+    if(this.audioElement.duration !== NaN){
```

Comment:
a stylistic consideration that makes code easier to read is:

1. eliminate the else in a case like this
2. make you conditionals "positive", e.g., "if it is a number..."

Reply:
Thanks for pointing this out, a good consideration to keep in mind. In this case, the conditional is actually no longer needed, as other code now takes care of the duration being returned properly.

Reply:
I'm a big fan of no conditionals so 🎉 !!!